compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1398,10 +1401,41 @@ fn report_ice(
         dcx.emit_err(session_diagnostics::Ice);
     }
 
-    if using_internal_features.load(std::sync::atomic::Ordering::Relaxed) {
-        dcx.emit_note(session_diagnostics::IceBugReportInternalFeature);
+    use time::ext::NumericalDuration;
+
+    // Try to hint user to update nightly if applicable when reporting an ICE.
+    // Attempt to calculate when current version was released, and add 12 hours
+    // as buffer. If the current version's release timestamp is older than
+    // the system's current time + 24 hours + 12 hours buffer if we're on
+    // nightly.
+    if let Some("nightly") = option_env!("CFG_RELEASE_CHANNEL")
+        && let Some(version) = option_env!("CFG_VERSION")
+        && let Some(ver_date_str) = option_env!("CFG_VER_DATE")
```

Comment:
I'm *assuming* these must be set by bootstrap for nightly builds?

Reply:
good question 😆 can do some investigation? If it turns out inconclusive, we'll just land it and check the next nightly 🙃

Reply:
> good question 😆 can do some investigation? If it turns out inconclusive, we'll just land it and check the next nightly 🙃

My assumption is based on

https://github.com/rust-lang/rust/blob/53ed660d47feb01055483fe81b628d5ef9705dbd/compiler/rustc_driver_impl/src/lib.rs#L846-L862

and 

https://github.com/rust-lang/rust/blob/53ed660d47feb01055483fe81b628d5ef9705dbd/compiler/rustc_session/src/errors.rs#L39-L49

so I'm fairly certain they are available for nightly compilers.